src/main/java/com/ukubuka/core/parser/impl/UkubukaXMLParser.java

```diff
@@ -41,4 +53,44 @@ public FileContents parseFile(final String completeFileName,
     public String getParserInfo() {
         return this.getClass().getSimpleName();
     }
+    
+    /**
+     * Read With Options
+     * 
+     * @param completeFileName
+     * @return File Content
+     * @throws ParserException
+     */
+    private List<String> readWithOptions(final String completeFileName,
+                                         Map<String, Object> flags) throws ParserException {
+        boolean withHeader = null == flags
+                .get(ExtractFlags.FILE_CONTAINS_HEADER.getFlag()) || (boolean) flags.get(
+                ExtractFlags.FILE_CONTAINS_HEADER.getFlag());
+        SupportedSource source = null == flags
+                .get(ExtractFlags.SOURCE.getFlag()) ? SupportedSource.FILE
+                        : SupportedSource.getSource((String) flags
+                                .get(ExtractFlags.SOURCE.getFlag()));
+        List<String> fileContents = readWithOptions(source, completeFileName);
+        return withHeader ? fileContents : super.appendHeader(fileContents);
+    }
+    
+    /**
+     * Read With Options
+     * 
+     * @param completeFileName
+     =* @return File Content
+     * @throws ParserException
+     */
+    private List<String> readWithOptions(final SupportedSource source,
+            final String completeFileName) throws ParserException {
+        try {
+            String[] fileContents = super.getReader()
+                    .readXMLAsString(source, completeFileName)
```

Comment:
All the XML Parsing logic should reside in this class. As per Single Responsibility Principle, the Reader class is only responsible for reading a file from disk and giving you the file contents. It makes no sense to put XML specific login in Reader. This defeats the whole purpose of Parser class(es).

Reply:
Done.

Reply:
Correct! Moved the parsing logic to XML Parser.